cinema/views.py

```diff
@@ -103,6 +106,32 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    @action(detail=True, methods=["post"], parser_classes=[MultiPartParser])
+    def upload_image(self, request, pk=None):
+        try:
+            movie = self.get_object()
+        except Movie.DoesNotExist:
+            return Response(
+                {"detail": "Movie not found."},
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        image = request.FILES.get("image")
+        if not image:
+            return Response(
+                {"detail": "No image file provided."},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        movie.image = image
+        movie.save()
+        return Response(
+            {
+                "detail": "Image uploaded successfully.",
+                "image": movie.image.url
+            }
+        )
```

Comment:
The `upload_image` action in `MovieViewSet` handles image uploads for movies. Ensure that the `MultiPartParser` is correctly configured to handle file uploads, and that the image field in the `Movie` model is set to accept file uploads as expected.